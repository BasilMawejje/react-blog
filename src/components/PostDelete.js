import React from 'react'
import { connect } from 'react-redux';
import { fetchPost, deletePost } from '../actions'

class PostDelete extends React.Component {
    componentDidMount() {
        this.props.fetchPost(this.props.match.params.id)
    }

    render() {
        if(!this.props.post) {
            return <div>Loading...</div>
        }

        const { title } = this.props.post

        return (
            <div>
                <div>
                    <h1>Are you sure you want to delete this post?</h1>
                    <h5>{title}</h5>
                    <button onClick={() => this.props.deletePost(this.props.post.id)} className='ui button negative'>Delete</button>
                </div>
            </div>
        );
    };
}

const mapStateToProps = (state, ownProps) => {
    const { id } = ownProps.match.params;
    return { post: state.posts[id] }
}

export default connect(mapStateToProps, { fetchPost, deletePost })(PostDelete);