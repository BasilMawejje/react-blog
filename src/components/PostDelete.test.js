import React from 'react';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux'
import { shallowToJson } from 'enzyme-to-json';

import PostDelete from './PostDelete'

configure({ adapter: new Adapter() });

const mockStore = configureStore();
const store = mockStore({posts: 
    [{ id: 1, title: 'Some post', body: 'lorem ipsum ...' }]
});

describe('PostDelete', () => {
    let wrapper; 

    beforeEach(() => {
        wrapper = shallow(
            <Provider store={store}>
                <PostDelete />
            </Provider>
        ).dive();
    });

    it('renders properly', () => { 
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
})