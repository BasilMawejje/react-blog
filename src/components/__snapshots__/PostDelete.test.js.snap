// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDelete renders properly 1`] = `<Connect(PostDelete) />`;
